pydantic/types.py

```diff
@@ -2393,9 +2393,11 @@ def __get_pydantic_json_schema__(
         return field_schema
 
     def __get_pydantic_core_schema__(self, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        if isinstance(source(), bytes):
```

Comment:
`if not`?

Reply:
I feel like we should do something more like the following, to be more consistent with other types in this file...

```py
if cls is source:
    # used directly as a type
     schema = core_schema.bytes_schema()
else:
    schema = handler(source)
    _check_annotated_type(schema['type'], 'bytes', cls.__name__)

return core_schema.with_info_after_validator_function(
    function=self.decode,
    schema=schema,
    serialization=core_schema.plain_serializer_function_ser_schema(function=self.encode),
)
```

Reply:
Is the first condition possible? I don't think it makes sense to use `EncodedBytes` as is: `TypeAdapter(EncodedBytes(...))`

Reply:
maybe 

```
schema = handler(source)
 _check_annotated_type(schema['type'], 'bytes', self.__classs__.__name__)
```

is enough